game-state-service

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import json
+import traceback
+
+from gi.repository import Gio
+from gi.repository import GLib
+
+
+class GameState(object):
+
+    _DEFERRED_TIME = 250
+
+    def __init__(self):
+        self._state = {}
+        self._state_save_deferred_id = None
+        self._state_from_file()
+
+    def _state_get_file(self):
+        path = GLib.build_filenamev([GLib.get_user_data_dir(), 'state.json'])
+        return Gio.File.new_for_path(path)
+
+    def _state_from_file(self):
+        state_file = self._state_get_file()
+        try:
+            success, state_data, tag = state_file.load_contents(None)
+            self._state = json.loads(state_data.decode())
+        except (GLib.Error, json.decoder.JSONDecodeError):
+            traceback.print_exc()
+
+    def _state_to_file(self):
+        state_data = json.dumps(self._state,
+                                sort_keys=True,
+                                indent=2,
+                                separators=(',', ': '))
+        state_file = self._state_get_file()
+        try:
+            state_file.replace_contents(
+                bytes(state_data.encode()),
```

Comment:
`bytes` is unnecessary here, `encode()` already returns a `bytes`.

Reply:
This is gone now, since I am only using JSON-GLIb API.